manifests/init.pp

```diff
@@ -185,10 +185,16 @@
       path         => $path,
       alias        => "concat_${name}",
       source       => "${fragdir}/${concat_name}",
-      validate_cmd => $validate_cmd,
       backup       => $backup,
     }
 
+    # Only newer versions of puppet 3.x support the validate_cmd parameter
+    if $validate_cmd {
+      File[$name] {
+        validate_cmd => $validate_cmd,
+      }
+    }
```

Comment:
If the validate_cmd parameter is unconditionally added to the `file {}` resource above then pre-3.5 versions will throw an error, irrespective of whether it's undef or set.